libcnb/src/build.rs

```diff
@@ -16,7 +15,11 @@ pub struct BuildContext<B: Buildpack + ?Sized> {
     pub layers_dir: PathBuf,
     pub app_dir: PathBuf,
     pub buildpack_dir: PathBuf,
-    pub stack_id: StackId,
+    pub target_os: String,
+    pub target_arch: String,
+    pub target_arch_variant: Option<String>,
+    pub target_distro_name: Option<String>,
+    pub target_distro_version: Option<String>,
```

Comment:
I think some rustdocs for these fields (and the equivalents in `DetectContext`) would help buildpack maintainers with the transition from `stack_id` to them. I think we should include:
- a brief description of the field (could be taken from the spec)
- an example value
- where the value originates from in practice (eg is it from the metadata set in the build image, derived, or ..?)

Reply:
Added in [dfdd601](https://github.com/heroku/libcnb.rs/pull/773/commits/dfdd60124aa67fdb2376e55f2a12732b9b29a5c6).

When adding these doc strings, the amount of duplication for detect and build became too large and I moved all target information into a new `ContextTarget` struct and added a shared initialiser in [0104524](https://github.com/heroku/libcnb.rs/pull/773/commits/0104524b6a54d3f535bba46efdf0953276217a91).

Reply:
Thank you - these new rustdocs are great :-)